src/gdpr/index.ts

```diff
@@ -96,9 +97,11 @@ export const runOnConsent = async <T>(
   callback: () => Promise<T>,
   omitGdprConsent = false
 ): Promise<T> => {
+  let hasConsent = true;
   if (!omitGdprConsent) {
-    await waitForConsent(vendorIds);
+    hasConsent = await waitForConsent(vendorIds);
   }
+  if (!hasConsent) throw new Error('No Gdpr consent given');
```

Comment:
This is not an error, the user is free to not grant consent.

Reply:
It has to return a `Promise<T>` because of the async, or not return at all.
Apart from throwing, if we do not want to wait forever, the only way I can think to make it work is to change the signature to take a second callback to be called in case of no consent (which may or may not have the same type from the success callback).

Reply:
`waitForConsent` could run forever this is fine.

I think the second callback complicates things, as the use case on the consumer does not need this, it is expected to just not run, we could just resolve/return  and empty result - or maybe reject, but then that would need changes on the consumer too.

Reply:
Ok, if it can wait forever it is easy.
I had misunderstood that you didn't want it to run forever.
Will fix it then =)